framework/include/utils/MooseUtils.h

```diff
@@ -488,6 +488,18 @@ ExecFlagEnum getDefaultExecFlagEnum();
  */
 int stringToInteger(const std::string & input, bool throw_on_failure = false);
 
+/**
+ * Implode a vector of strings to create a single string
+ */
+template <typename T>
+std::string
+implodeString(const T & strings, const char * delim)
+{
+  std::ostringstream imploded;
+  std::copy(strings.begin(), strings.end(), std::ostream_iterator<std::string>(imploded, delim));
+  std::string result = imploded.str();
+  return result.erase(result.rfind(delim));
+}
```

Comment:
People keep trying to add this. It exists (probably at least twice in MOOSE).

https://github.com/idaholab/moose/blob/devel/framework/include/utils/Conversion.h#L76-L86